command/meta.go

```diff
@@ -427,3 +438,197 @@ type funcVar func(s string) error
+		Description: "View allocation details",
+	},
+	"var list": {
+		Path:        "/variables",
+		Description: "View Nomad variables",
+	},
+	"var list prefix": {
+		Path:        "/variables/path/:prefix",
+		Description: "View Nomad variables at this path",
+	},
+	"var get": {
+		Path:        "/variables/var/:path@:namespace",
+		Description: "View variable details",
+	},
+	"var put": {
+		Path:        "/variables/var/:path@:namespace",
+		Description: "View variable details",
+	},
+	"job dispatch": {
+		Path:        "/jobs/:dispatchID@:namespace",
+		Description: "View this job",
+	},
+	"eval list": {
+		Path:        "/evaluations",
+		Description: "View evaluations",
+	},
+	"eval status": {
+		Path:        "/evaluations?currentEval=:evalID",
+		Description: "View evaluation details",
+	},
+	"deployment status": {
+		Path:        "/jobs/:jobID/deployments",
+		Description: "View all deployments for this job",
+	},
+}
+
+func (m *Meta) formatUIHint(url string, description string) string {
+	if description == "" {
+		description = defaultHint
+	}
+
+	description = fmt.Sprintf("%s in the Web UI:", description)
+
+	// Basic version without colors
+	hint := fmt.Sprintf("%s%s %s", uiHintDelimiter, description, url)
+
+	// If colors are disabled, return basic version
+	_, coloredUi := m.Ui.(*cli.ColoredUi)
+	if m.noColor || !coloredUi {
+		return hint
+	}
+
+	return fmt.Sprintf("%[1]s%[2]s%[3]s%[4]s%[5]s %[6]s%[7]s%[8]s",
+		bold,
+		magenta,
+		uiHintDelimiter[1:], // "==> "
+		description,
+		resetter,
+		blue,
+		url,
+		resetter,
+	)
+}
+
+func (m *Meta) buildUIPath(route UIRoute, params map[string]string) (string, error) {
+	client, err := m.Client()
+	if err != nil {
+		return "", fmt.Errorf("error getting client config: %v", err)
+	}
+
+	path := route.Path
+	for k, v := range params {
+		path = strings.ReplaceAll(path, fmt.Sprintf(":%s", k), v)
+	}
+
+	return fmt.Sprintf("%s/ui%s", client.Address(), path), nil
+}
+
+func (m *Meta) showUIPath(ctx UIHintContext) (string, error) {
+	route, exists := CommandUIRoutes[ctx.Command]
+	if !exists {
+		return "", nil
+	}
+
+	url, err := m.buildUIPath(route, ctx.PathParams)
+	if err != nil {
+		return "", err
+	}
+
+	if ctx.OpenURL {
+		if err := util.OpenURL(url); err != nil {
+			m.Ui.Warn(fmt.Sprintf("Failed to open browser: %v", err))
+		}
+	}
+
+	if m.uiHintsDisabled() {
+		return "", nil
+	}
+
+	return m.formatUIHint(url, route.Description), nil
+}
+
+func (m *Meta) uiHintsDisabled() bool {
+	// Either the local env var is set to false,
+	// or the agent config is set to false nad the local config isn't set to true
+
+	// First check if the user/env var is set to false. If it is, return early.
+	if m.showCLIHints != nil && !*m.showCLIHints {
+		return true
+	}
+
+	// Next, check if the agent config is set to false. If it is, return early.
+	client, err := m.Client()
+	if err != nil {
+		return true
+	}
+
+	agent, err := client.Agent().Self()
+	if err != nil {
+		return true
+	}
```

Comment:
Addressing a comment made in our slack discussion: although this is making another API call, _most_ of our APIs end up making multiple API calls anyways so I don't feel like this is an enormous burden. And it'll drive us towards having some kind of local config file to govern the Nomad CLI too, which is something we've discussed a bunch previously.